### PR TITLE
Fix solverhandler for hand joint rotation

### PR DESCRIPTION
**Pull request by Troy-Ferrell**
## Overview
Cleans up solver code and adds documentation and example code for fixing hand joint targeting.

The hand joint has the forward vector go through the fingers which is generally not the desired vector when used with the solvers. To overcome this, the `AdditionalRotation` property can be used to fix the hand joint forward vector to actually point through the palm

## Changes
- Fixes: #7011

### CODE REVIEW

File: Documentation/README_Solver.md [hunk 1]
```diff
@@ -49,6 +49,17 @@ The *Tracked Target Type* property of the [`SolverHandler`](xref:Microsoft.Mixed
 ![Solver](Images/Solver/TrackedObjectType-Example.gif)  
 *Example of various properties associated with each TrackedTargetType*
 
+> [!IMPORTANT]
+> Most Solvers use the forward vector of the tracked transform target supplied by the `SolverHandler`. When using a *Hand Joint* Tracked Target Type, the forward vector of the palm joint may point through the fingers and not through the palm. This depends on the platform supplying the hand joint data. For input simulation and Windows Mixed Reality, it is the *up vector* that points up through the palm (i.e green vector is up, blue vector is forward).
```
Comment (thalbern) [inline Documentation/README_Solver.md]:
```suggestion
> Most solvers use the forward vector of the tracked transform target supplied by the `SolverHandler`. When using a *Hand Joint* tracked target type, the forward vector of the palm joint may point through the fingers and not through the palm. This depends on the platform supplying the hand joint data. For input simulation and Windows Mixed Reality, it is the *up vector* that points up through the palm (i.e green vector is up, blue vector is forward).
```

File: Documentation/README_Solver.md [hunk 2]
```diff
@@ -49,6 +49,17 @@ The *Tracked Target Type* property of the [`SolverHandler`](xref:Microsoft.Mixed
...
+> 
+> ![Solver](Images/Solver/HandJoint_ForwardUpVectors.png)
+> 
+> To overcome this, update the *Additional Rotation* property on the `SolverHandler` to **<90, 0, 0>**. This will ensure the forward vector supplied to Solvers is pointing through the palm and outward away from the hand.
```
Comment (thalbern) [inline Documentation/README_Solver.md]:
```suggestion
> To overcome this, update the *Additional Rotation* property on the `SolverHandler` to **<90, 0, 0>**. This will ensure the forward vector supplied to solvers is pointing through the palm and outward away from the hand.
```

File: Documentation/README_Solver.md [hunk 3]
```diff
@@ -49,6 +49,17 @@ The *Tracked Target Type* property of the [`SolverHandler`](xref:Microsoft.Mixed
...
+>
+> ![Solver](Images/Solver/SolverHandler_AdditionalRotation.png)
+>
+> Alternatively, use the *Controller Ray* Tracked Target Type to get similar behavior for pointing with hands.
```
Comment (thalbern) [inline Documentation/README_Solver.md]:
```suggestion
> Alternatively, use the *Controller Ray* tracked target type to get similar behavior for pointing with hands.
```

File: Assets/MixedRealityToolkit.Examples/Demos/Solvers/Scripts/SolverExampleManager.cs [hunk 4]
```diff
@@ -96,6 +98,13 @@ private void RefreshSolverHandler()
             if (handler != null)
             {
                 handler.TrackedTargetType = TrackedType;
+
+#if WINDOWS_UWP || UNITY_EDITOR
+                // If using input simulation in editor or on Windows Mixed Reality platform, forward vector points through the fingers, not the palm.
+                // Apply additional rotation to correct this so that forward will be through the palm and outward
+                handler.AdditionalRotation = (TrackedType == TrackedObjectType.HandJoint) ? HandJointRotationFix : Vector3.zero;
```
Comment (wiwei) [inline Assets/MixedRealityToolkit.Examples/Demos/Solvers/Scripts/SolverExampleManager.cs]: A bunch of random thoughts here:

I think I'm having a slight mental disconnect on this because these solver things are explicitly trying to track to a specific joint, and currently the code is doing exactly that - this fix is kinda changing the mechanics of what joint tracking is (i.e. instead of honoring the values from the platform, instead being like "actually, we want palm outward, let's try to massage the data to conform to that requirement). If we actually want "palm outward tracking" it feels like this should be a specific concept that we surface (i.e. maybe we have "meta joints" which share the the same location, but are maybe called "PalmMetaJointOutward" where we can then do some cross products and figure out the orientation ourselves).

Otherwise, we end up having to do per-platform fixes (and potentially re-fixes as the platform changes things or data gets "better").

That being said, I think this change is still better than no change (up going through the fingers doesn't make sense to anyone) so I will step off of my soapbox(?) now.
tl;dr if we

Comment (Troy-Ferrell) [reply Assets/MixedRealityToolkit.Examples/Demos/Solvers/Scripts/SolverExampleManager.cs]: All very good points. I was having a similar discussion in my head when looking at this issue/change. I originally looked at changing code in the solver(s) or SolverHandler. But I didn't want to have to deal with a per-platform edge case like this at the core.

Thus, I understand your concerns but I don't understand why you have them for this class, `SolverExampleManager`?

As a platform, we are providing joint tracking as is. It would be the app developer that is modifying the transform themselves by configuring the AdditionalRotation property.

Thus, the responsibility is on them to do so and handle it per-platform as desired

Comment (wiwei) [reply Assets/MixedRealityToolkit.Examples/Demos/Solvers/Scripts/SolverExampleManager.cs]: Ah oh no, I misread the SolverExampleManager as just SolverManager or otherwise thought of it as more foundational than it is.

**Comment by Troy-Ferrell** [comment]
@thalbern the moving obsolete fields to bottom is just kind of a personal preference. If you guys have any other suggestion or guideline to add, happy to follow
